regexing/src/main/java/Extraction.java

```diff
@@ -24,6 +32,12 @@
      */
     public List<String> extractUrls(String input){
         List<String> urls = new ArrayList<String>();
+        Pattern p = Pattern.compile("(?:href=\")([\\w:/.]+)");
```

Comment:
I am a little confused why you did a non-capture group around the href.  Non-capture groups are used to act on a sequence as a whole.  You create this one and do nothing with it.